proxy.go

```diff
@@ -226,12 +268,24 @@ func main() {
 		logger().Fatalf("failed to dial channel(%q): %s", muxAddr, err)
 		return
 	}
-	defer servConn.Close()
+	defer func() {
+		if servConn != nil {
+			servConn.Close()
+			servConn = nil
+		}
+	}()
 
 	results := make(chan error)
-	err = serve(servConn, "unix", listenAddr, results)
+	l, err := serve(servConn, "unix", listenAddr, results)
 	if err != nil {
 		logger().Fatal(err)
+		return
+	}
+
+	// Blocking function waiting for a SIGTERM signal.
```

Comment:
I'd remove this comment (as it's more duplication of `SIGTERM`) and just add a comment to the signal handler function that it's blocking.

Reply:
Makes sense